core/src/main/java/org/elasticsearch/discovery/zen/ZenDiscovery.java

```diff
@@ -35,6 +38,7 @@
 import org.elasticsearch.cluster.routing.allocation.AllocationService;
 import org.elasticsearch.cluster.routing.allocation.RoutingAllocation;
 import org.elasticsearch.cluster.service.ClusterService;
+import org.elasticsearch.cluster.service.DuplicateClusterStateUpdateTaskException;
```

Comment:
I pushed cf41ff445d15b3f1fc4a4a38a1acd996b1e5e662. I think that it should be an `IllegalStateException` though.